src/useValidationObserver/useValidationObserver.spec.ts

```diff
@@ -0,0 +1,133 @@
+import { mount } from '@vue/test-utils'
+import { describe, it, expect, vi } from 'vitest'
+import { nextTick, ref } from 'vue'
+import { useValidation } from '@/useValidation'
+import { useValidationObserver } from '@/useValidationObserver'
+
+describe('useValidationObserver', () => {
+  it('should observe a single useValidation', async () => {
+    const validationRule = vi.fn().mockResolvedValue(true)
+    const wrapper = mount({
+      setup() {
+        const { validate } = useValidationObserver()
+        useValidation(ref(0), validationRule)
+        return { validate }
+      },
+    })
+
+    const result = await wrapper.vm.validate()
+
+    expect(result).toBe(true)
+    expect(validationRule).toHaveBeenCalled()
+  })
+
+  it('should observe multiple useValidations', async () => {
+    const validationRule1 = vi.fn().mockResolvedValue(true)
+    const validationRule2 = vi.fn().mockResolvedValue(true)
+    const wrapper = mount({
+      setup() {
+        const { validate } = useValidationObserver()
+        useValidation(ref(0), validationRule1)
+        useValidation(ref(0), validationRule2)
+
+        return { validate }
+      },
+    })
+
+    const result = await wrapper.vm.validate()
+
+    expect(result).toBe(true)
+    expect(validationRule1).toHaveBeenCalled()
+    expect(validationRule2).toHaveBeenCalled()
+  })
+
+  it('should observe multiple useValidations and be invalid if any does not pass', async () => {
+    const validationRule1Spy = vi.fn().mockResolvedValue(true)
+    const validationRule2Spy = vi.fn().mockResolvedValue('Number must be greater than 0')
```

Comment:
You can also just return `false` from a validation rule and it'll give a generic error. So you probably don't need to mock error strings in these unless you're testing the strings specifically.